tools/src/main/java/org/apache/kafka/tools/streams/StreamsGroupCommand.java

```diff
@@ -368,7 +368,11 @@ static boolean isGroupStateValid(GroupState state, int numRows) {
         private static Set<TopicPartition> getTopicPartitions(List<StreamsGroupMemberAssignment.TaskIds> taskIds, StreamsGroupDescription description) {
             Map<String, List<String>> allSourceTopics = new HashMap<>();
             for (StreamsGroupSubtopologyDescription subtopologyDescription : description.subtopologies()) {
-                allSourceTopics.put(subtopologyDescription.subtopologyId(), subtopologyDescription.sourceTopics());
+                List<String> topics = new ArrayList<>(subtopologyDescription.sourceTopics());
+                if (!subtopologyDescription.repartitionSourceTopics().isEmpty()) {
```

Comment:
Excuse me, is it necessary to add an empty check? It seems to me that removing the empty check has no side effect, and it can avoid an extra map copy.